### PR TITLE
Add testing for Bazel browser setup

### PR DESCRIPTION
**Pull request by devversion**
See individual commits. A lot of things were coupled as some issues just surfaced due to cache invalidation and new CI executors. All of these issues are confirmed locally and they are reasonable (so nothing was flakiness!)

### CODE REVIEW

**Comment by devversion** [comment]
@josephperrott this would fix the firefox shasum error in FW

File: bazel/browsers/update-script/browser-artifact.ts [hunk 1]
```diff
@@ -37,14 +36,20 @@ export class BrowserArtifact {
...
+export function detectArtifactExtension(filePath: string) {
   let tmpPath: string = filePath;
   let extension: string = '';
   let currentPart: string = '';
 
   // Iterate from the end of the path, finding the largest possible
   // extension substring, accounting for cases like `a/b.tmp/file.tar.gz`.
   while ((currentPart = path.extname(tmpPath)) !== '') {
+    // An extension needs to be a continuous set of alphanumeric characters. This is a strict
+    // requirement, but sufficient for our needs and helps making the determinationmore proof.
```
Comment (josephperrott) [inline bazel/browsers/update-script/browser-artifact.ts:48]: I am not sure what you mean at the end of this sentence

**Review by josephperrott — APPROVED** [review]
Overall LGTM

Just a couple questions.

File: bazel/browsers/update-script/browser-artifact.ts (hunk 1 above)
Comment (devversion) [reply bazel/browsers/update-script/browser-artifact.ts:48]: updated the comment and added an example

File: .circleci/config.yml [hunk 2]
```diff
@@ -162,6 +202,8 @@ workflows:
   default_workflow:
     jobs:
       - test
+      - test_macos_x64
```
[the remark this answers is not in the document]
Comment (devversion) [reply .circleci/config.yml:205]: I think we wanted to run them in main only. I just forgot about it due to testing.

**Pull request merged**